src/mono/wasm/runtime/startup.ts

```diff
@@ -486,19 +478,19 @@ async function instantiate_wasm_module(
 
         mono_log_debug("instantiate_wasm_module done");
 
-        if (runtimeHelpers.loadedMemorySnapshot) {
+        if (runtimeHelpers.loadedMemorySnapshotSize) {
             try {
                 const wasmMemory = Module.getMemory();
 
                 // .grow() takes a delta compared to the previous size
-                wasmMemory.grow((memorySize! - wasmMemory.buffer.byteLength + 65535) >>> 16);
+                wasmMemory.grow((runtimeHelpers.loadedMemorySnapshotSize! - wasmMemory.buffer.byteLength + 65535) >>> 16);
```

Comment:
What happens if the snapshot is somehow smaller than the current memory buffer? Is that possible? MDN doesn't say what a negative argument to grow would do, but I assume it would fail.

Reply:
Since you're doing >>> instead of >> I think it might underflow and cause us to try and grow by a huge amount?

Reply:
I believe that we don't ever store memory smaller than the initial size, because I think emscripten never shrinks it.

Reply:
And if there was new version of the .wasm it would invalidate the hash and skip this path.